controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go

```diff
@@ -234,6 +239,23 @@ type RemediationStrategy struct {
 	MinHealthyPeriod *metav1.Duration `json:"minHealthyPeriod,omitempty"`
 }
 
+// MachineNamingStrategy allows changing the naming pattern used when creating Machines.
+// InfraMachines & KubeadmConfigs will use the same name as the corresponding Machines.
+type MachineNamingStrategy struct {
+	// Template defines the template to use for generating the names of the Machine objects.
+	// If not defined, it will fallback to `{{ .cluster.name }}-{{ .kubeadmcontrolplane.name }}-{{ .random }}`.
+	// If the generated name string exceeds 63 characters, it will be trimmed to 58 characters and will
+	// get concatenated with a random suffix of length 5.
+	// Length of the template string must not exceed 93 characters.
+	// The template allows the following variables `.cluster.name`, `.kubeadmcontrolplane.name` and `.random`.
+	// The variable `.cluster.name` retrieves the name of the cluster object that owns the Machines being created.
+	// The variable `.kubeadmcontrolplane.name` retrieves the name of the KubeadmControlPlane object that owns the Machines being created.
+	// The variable `.random` is substituted with random alphanumeric string, without vowels, of length 5.
```

Comment:
Is it on purpose that we want to allow users to specify where the random part should be?

What happens if the random part is missed and every name is the same?

Reply:
yes, this is an issue, this errors out when trying to create 2nd CP with the same name. What we can do is if the user misses it, it will be added at the end by default, and I can update the PR. 
What do you think?

Reply:
Have we considered to allow the prefix to be customized, but not the random part, which is always at the end?

Reply:
regarding random always being at the end: https://github.com/kubernetes-sigs/cluster-api/pull/11123#discussion_r1777527638

> Another case would be if folks have the (admittedly strange) use case that the random part shouldn't be at the end of the name (there was one (!) person asking for it in Slack, they wanted a random prefix for some reason, xref: [kubernetes.slack.com/archives/C8TSNPY4T/p1724389852680579](https://kubernetes.slack.com/archives/C8TSNPY4T/p1724389852680579))

Reply:
I have updated the PR, if user misses the `.random` by default it will be added at the end.

Reply:
Do we really want inconsistencies like this between various of our namimgStrategy fields?

Reply:
What would be okay is to validate that random is used and otherwise return a validation error

Reply:
done